ports/esp32/machine_can.h

```diff
@@ -0,0 +1,137 @@
+
+typedef struct _machine_can_obj_t {
+    mp_obj_base_t base;
+    machine_can_config_t *config;
+} machine_can_obj_t;
+
+//Internal functions
+STATIC can_state_t _machine_hw_can_get_state();
+
+//Functions signature definition
+mp_obj_t machine_hw_can_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args);
+STATIC mp_obj_t machine_hw_can_init_helper(const machine_can_obj_t *self, size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args);
+STATIC mp_obj_t machine_hw_can_init(size_t n_args, const mp_obj_t *args, mp_map_t *kw_args);
+STATIC mp_obj_t machine_hw_can_deinit(const mp_obj_t self_in);
+STATIC void machine_hw_can_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind);
+
+STATIC mp_obj_t machine_hw_can_recv(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args);
+STATIC mp_obj_t machine_hw_can_send(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args);
+STATIC mp_obj_t machine_hw_can_get_tec(mp_obj_t self_in);
+STATIC mp_obj_t machine_hw_can_get_rec(mp_obj_t self_in);
+STATIC mp_obj_t machine_hw_can_get_tx_waiting_messages(mp_obj_t self_in);
+STATIC mp_obj_t machine_hw_can_get_rx_waiting_messages(mp_obj_t self_in);
+STATIC mp_obj_t machine_hw_can_get_state(mp_obj_t self_in);
+STATIC mp_obj_t machine_hw_can_clear_tx_queue(mp_obj_t self_in);
+STATIC mp_obj_t machine_hw_can_clear_rx_queue(mp_obj_t self_in);
+
+//Python function declarations
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_deinit_obj, machine_hw_can_deinit);
+MP_DEFINE_CONST_FUN_OBJ_KW(machine_hw_can_init_obj, 4, machine_hw_can_init);
+MP_DEFINE_CONST_FUN_OBJ_KW(machine_hw_can_send_obj, 3, machine_hw_can_send);
+MP_DEFINE_CONST_FUN_OBJ_KW(machine_hw_can_recv_obj, 0, machine_hw_can_recv);
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_get_tec_obj, machine_hw_can_get_tec);
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_get_rec_obj, machine_hw_can_get_rec);
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_get_tx_waiting_messages_obj, machine_hw_can_get_tx_waiting_messages);
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_get_rx_waiting_messages_obj, machine_hw_can_get_rx_waiting_messages);
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_get_state_obj, machine_hw_can_get_state);
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_clear_tx_queue_obj, machine_hw_can_clear_tx_queue);
+MP_DEFINE_CONST_FUN_OBJ_1(machine_hw_can_clear_rx_queue_obj, machine_hw_can_clear_rx_queue);
+
+//Python objects list declaration
+STATIC const mp_rom_map_elem_t machine_can_locals_dict_table[] = {
+    //CAN_ATTRIBUTES
+    { MP_ROM_QSTR(MP_QSTR___name__), MP_ROM_QSTR(MP_QSTR_CAN) },
+    //CAN_FUNCTIONS
+    { MP_ROM_QSTR(MP_QSTR_init), MP_ROM_PTR(&machine_hw_can_init_obj) },
+    { MP_ROM_QSTR(MP_QSTR_deinit), MP_ROM_PTR(&machine_hw_can_deinit_obj) },
+    { MP_ROM_QSTR(MP_QSTR_send), MP_ROM_PTR(&machine_hw_can_send_obj)},
+    { MP_ROM_QSTR(MP_QSTR_recv), MP_ROM_PTR(&machine_hw_can_recv_obj)},
+    { MP_OBJ_NEW_QSTR(MP_QSTR_get_tec), MP_ROM_PTR(&machine_hw_can_get_tec_obj)},
```

Comment:
These four methods don't exist in STM32 -- instead it just has `any()`.

If your application needs these counters, could it possibly be collapsed to a single method `stats` returning a tuple or something?